src/components/ExchangePage/index.tsx

```diff
@@ -311,48 +314,39 @@ function ExchangePage({ sendingInput = false, history, params }: ExchangePagePro
 
   function getSwapType(): SwapType {
     if (tradeType === TradeType.EXACT_INPUT) {
-      if (tokens[Field.INPUT] === WETH[chainId]) {
+      if (tokens[Field.INPUT].equals(WETH[chainId])) {
         return SwapType.EXACT_ETH_FOR_TOKENS
-      } else if (tokens[Field.OUTPUT] === WETH[chainId]) {
+      } else if (tokens[Field.OUTPUT].equals(WETH[chainId])) {
         return SwapType.EXACT_TOKENS_FOR_ETH
       } else {
         return SwapType.EXACT_TOKENS_FOR_TOKENS
       }
     } else if (tradeType === TradeType.EXACT_OUTPUT) {
-      if (tokens[Field.INPUT] === WETH[chainId]) {
+      if (tokens[Field.INPUT].equals(WETH[chainId])) {
         return SwapType.ETH_FOR_EXACT_TOKENS
-      } else if (tokens[Field.OUTPUT] === WETH[chainId]) {
+      } else if (tokens[Field.OUTPUT].equals(WETH[chainId])) {
         return SwapType.TOKENS_FOR_EXACT_ETH
       } else {
         return SwapType.TOKENS_FOR_EXACT_TOKENS
       }
     }
   }
 
-  function calculateSlippageAmount(value: TokenAmount): JSBI[] {
-    if (value && value.raw) {
-      const offset = JSBI.divide(JSBI.multiply(JSBI.BigInt(allowedSlippage), value.raw), JSBI.BigInt(10000))
-      return [JSBI.subtract(value.raw, offset), JSBI.add(value.raw, offset)]
-    }
-    return null
-  }
-
-  const slippageAdjustedAmounts: { [field in Field]: TokenAmount } = {
-    [Field.INPUT]:
-      Field.INPUT === independentField
-        ? parsedAmounts[Field.INPUT]
-        : calculateSlippageAmount(parsedAmounts[Field.INPUT])?.[0] &&
-          new TokenAmount(tokens[Field.INPUT], calculateSlippageAmount(parsedAmounts[Field.INPUT])?.[1]),
-    [Field.OUTPUT]:
-      Field.OUTPUT === independentField
-        ? parsedAmounts[Field.OUTPUT]
-        : calculateSlippageAmount(parsedAmounts[Field.OUTPUT])?.[0] &&
-          new TokenAmount(tokens[Field.INPUT], calculateSlippageAmount(parsedAmounts[Field.OUTPUT])?.[0])
+  const slippageAdjustedAmounts: { [field: number]: TokenAmount } = {
+    [independentField]: parsedAmounts[independentField],
+    [dependentField]: parsedAmounts[dependentField]
+      ? tradeType === TradeType.EXACT_INPUT
+        ? new TokenAmount(
+            tokens[dependentField],
+            calculateSlippageAmount(parsedAmounts[dependentField], allowedSlippage)[0]
```

Comment:
Can we add a unit test to this? Feel like it should be in SDK, something like `Trade#minimumAmountOut(allowedSlippage: Percent): TokenAmount` and `Trade#maximumAmountIn(allowedSlippage: Percent): TokenAmount`

For exact out/in trades the above methods respectively would just return output/input

Reply:
https://github.com/Uniswap/uniswap-sdk/pull/23

Reply:
added unit tests, agreed that it should be in the SDK!